test/media-query-provider.js

```diff
@@ -131,4 +131,29 @@ describe('<MediaQueryProvider />', () => {
       }).to.not.throw();
     });
   });
+
+  if (React.Fragment) {
```

Comment:
@gricard Your tests can be improved. I felt your pain when you tried to check if it has support for `React.Fragment` or not. Since this package is using `React v15` this test will check for the `else` condition always. This is one option to check your condition in the tests, without `if/else`.


```js

  context('when React.Fragment is available', () => {
    before(() => {
      Object.defineProperty(React, 'Fragment', {
        configurable: true,
        value: React.createClass({
          render: function() {
            return React.createElement(
              'span',
              {className: 'wrapper'},
              Array.isArray(this.props.children) ? 
                this.props.children.map((el) => <span>{el}</span>) :
                this.props.children
            );
          }
        }),
      });
    });

    after(() => {
      Reflect.deleteProperty(React, 'Fragment');
    });

    it('should use React.Fragment component', () => {
      const fragmentChildren = [
        <p>Test123</p>,
        <p>Test123</p>,
      ];
      const component = mount(
        <MediaQueryProvider>
          <fragmentChildren />
        </MediaQueryProvider>,
      );
      expect(component.find('span').is('span')).to.eql(true);
    });
  });

  context('when React.Fragment is not available', () => {
    it('should render a div', () => {
      const component = mount(
        <MediaQueryProvider>
          <p>Test123</p>
        </MediaQueryProvider>,
      );
      expect(component.find('div').is('div')).to.eql(true);
    });
  });
```

In that case, we are adding `span`, but we are covering the cases with `Fragment` or not since `MediaQueryProvider` is respective using `React.Fragment` when and if it exists.

Reply:
I think this is a decent solution for now - even if we update to react 16, still cant test fragments correctly until enzyme supports them

Reply:
@willmendesneto shouldn't that first test use React.Children.map instead of this.props.children.map?

Reply:
I haven't tried using `React.Children`, but I think that shouldn't work in this case since you need to check the children for the mocked component. I checked this code and it's checking properly your changes

Reply:
@willmendesneto Aside from this warning it looks good. I'm committing that change. 

```
Warning: value: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.
```